kvgraph/graph.go

```diff
@@ -43,25 +50,30 @@ func (kgdb *KVInterfaceGDB) AddVertex(vertexArray []*gripql.Vertex) error {
 		}
 	}
 
-	err := kgdb.kvg.kv.Update(func(tx kvi.KVTransaction) error {
+	dataChan := make(chan *kvAddData, 100)
+	go func() {
 		for _, vertex := range vertexArray {
 			d, err := proto.Marshal(vertex)
-			if err != nil {
-				return err
-			}
 			k := VertexKey(kgdb.graph, vertex.Gid)
-			err = tx.Set(k, d)
-			if err != nil {
-				return err
+			if err == nil {
+				doc := map[string]interface{}{kgdb.graph: vertexIdxStruct(vertex)}
+				dataChan <- &kvAddData{key: k, value: d, vertex: vertex, doc: doc}
 			}
-			doc := vertexIdxStruct(vertex)
-			err = kgdb.kvg.idx.AddDocTx(tx, vertex.Gid, map[string]interface{}{kgdb.graph: doc})
-			if err != nil {
-				return err
+		}
+		close(dataChan)
+	}()
+
+	err := kgdb.kvg.kv.Update(func(tx kvi.KVTransaction) error {
+		var anyErr error
+		for kv := range dataChan {
+			if err := tx.Set(kv.key, kv.value); err != nil {
+				anyErr = err
+			} else {
+				kgdb.kvg.idx.AddDocTx(tx, kv.vertex.Gid, kv.doc)
```

Comment:
Check the error returned by this method as well. 

Also since we aren't breaking the loop on error, maybe they should be logged?